extensions/wikia/PaidAssetDrop/js/PaidAssetDrop.desktop.js

```diff
@@ -0,0 +1,138 @@
+define('ext.wikia.PaidAssetDrop', [
+	'jquery',
+	'wikia.log',
+	'wikia.window'
+], function ($, log, win) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.PaidAssetDrop',
+		articleContentId = '#mw-content-text',
+		assetArticleName = 'MediaWiki:PAD_desktop.html',
+		apiEntryPoint = 'api.php?action=query&prop=revisions&rvlimit=1&rvprop=content&format=json&titles=';
+
+	log('Paid Asset Drop enabled', 'debug', logGroup);
+
+	function isValidDate(dateString) {
+		return !isNaN(Date.parse(dateString));
+	}
+
+	function isConfigValid() {
+		if (typeof win.wgPADConfig === 'undefined') {
+			log('wgPADConfig is undefined', 'debug', logGroup);
+			return false;
+		}
+
+		if (!win.wgPADConfig[0]) {
+			log('No start date set', 'debug', logGroup);
+			return false;
+		}
+
+		if (!win.wgPADConfig[1]) {
+			log('No start date set', 'debug', logGroup);
+			return false;
+		}
+
+		if (!isValidDate(win.wgPADConfig[0])) {
+			log('Start date invalid', 'debug', logGroup);
+			return false;
+		}
+
+		if (!isValidDate(win.wgPADConfig[1])) {
+			log('End date invalid', 'debug', logGroup);
+			return false;
+		}
+
+		return true;
+	}
+
+	function isTodayValid() {
+		var today, start, end;
+
+		if(!isConfigValid()) {
+			return false;
+		}
+
+		start = new Date(win.wgPADConfig[0]);
+		end = new Date(win.wgPADConfig[1]);
+		today = new Date();
+
+		log('PAD start date: ' + start, 'debug', logGroup);
+		log('PAD end date: ' + end, 'debug', logGroup);
+		log('PAD today is: ' + today, 'debug', logGroup);
+
+		if (today.getTime() < start.getTime()) {
+			log('PAD disabled: it is too early', 'debug', logGroup);
+			return false;
+		}
+
+		if (today.getTime() > end.getTime()) {
+			log('PAD disabled: it is too late', 'debug', logGroup);
+			return false;
+		}
+
+		log('PAD enabled', 'debug', logGroup);
+		return true;
+	}
+
+	function fetchPadContent(response) {
+		var page, revision;
+
+		if (response.query && response.query.pages && Object.keys) {
+			page = response.query.pages[Object.keys(response.query.pages)[0]];
```

Comment:
How about an optimistic long access with lots of "." in between wrapped in try/catch?